### PR TITLE
allow props in mountComponent

### PR DESCRIPTION
**Pull request by robinlej**

### CODE REVIEW

File: addons/website/static/src/core/website_core.js [hunk 1]
```diff
@@ -62,12 +62,12 @@ class WebsiteCore {
             };
             this.owlApp = new App(null, appConfig);
         }
-        const root = this.owlApp.createRoot(C, { props: null, env: this.env });
+        const root = this.owlApp.createRoot(C, { props, env: this.env });
```
Comment (bso-odoo) [inline addons/website/static/src/core/website_core.js:65]: Why not forward the `props` parameter ?

File: addons/website/static/tests/core/interaction.test.js [hunk 2]
```diff
@@ -1455,7 +1455,6 @@ describe("dynamic attributes", () => {
 });
 
 describe("components", () => {
-
     test("can insert a component with t-component", async () => {
```
Comment (bso-odoo) [inline addons/website/static/tests/core/interaction.test.js:1458]: Do you also want to cover `t-component` ?

File: addons/website/static/src/core/interaction.js [hunk 3]
```diff
@@ -266,8 +266,12 @@ export class Interaction {
...
     }
 
-    mountComponent(el, C) {
-        this.__colibri__.mountComponent([el], C);
+    /**
+     * @param {HTMLElement} el
+     * @param {import("@odoo/owl").Component} C
+     * @param {Object|null} props
+     */
+    mountComponent(el, C, props) {
```
Comment (ged-odoo) [inline addons/website/static/src/core/interaction.js]:
```js
    mountComponent(el, C, props = null) {
```

File: addons/website/static/src/core/colibri.js [hunk 4]
```diff
@@ -185,7 +185,11 @@ export class Colibri {
             } else if (directive === "t-out") {
                 this.tOuts.push([nodes, value]);
             } else if (directive === "t-component") {
-                this.mountComponent(nodes, value);
+                if (!!value.prototype) {
+                    this.mountComponent(nodes, value);
+                } else {
+                    this.mountComponent(nodes, ...value());
+                }
```
Comment (robinlej) [inline addons/website/static/src/core/colibri.js]: Don't know about that. Maybe we can just keep the `() => ([Component, props])` syntax

File: addons/website/static/src/core/colibri.js [hunk 5]
```diff
@@ -185,7 +185,12 @@ export class Colibri {
             } else if (directive === "t-out") {
                 this.tOuts.push([nodes, value]);
             } else if (directive === "t-component") {
-                this.mountComponent(nodes, value);
+                const { Component } = odoo.loader.modules.get("@odoo/owl");
+                if (Component.isPrototypeOf(value)) {
```
Comment (ged-odoo) [inline addons/website/static/src/core/colibri.js:189]:
```js
value.prototype instanceof Component
```

**Pull request merged**